concepts/pointers/about.md

```diff
@@ -1,3 +1,154 @@
+
+```go
+var a int
+```
+
+When we declare a variable, Go has to find a place in memory to store its value. This is largely abstracted from us — when we need to fetch the value stored in that piece of memory, we can just refer to it by the variable name.
+
+For instance, when we write `a + 2`, we are effectively fetching the value stored in the memory associated with the variable `a` and adding 2 to it.
+
+Similarly, when we need to change the value in the piece of memory of `a`, we can use the variable name to do an assignment:
+
+```go
+a = 3
+```
+
+The piece of memory that is associated with `a` will now will now be storing the value `3`.
+
+## Pointers
+
+While variables allow us to refer to values in memory, sometimes it's useful to know the **memory address** to which the variable is pointing. **Pointers** hold the memory addresses of those values. A pointer declaration looks like this:
+
+```go
+var p *int // 'p' contains the memory address of an integer
+```
+
+Here we declaring a variable `p` of type `*int`. The type `*int` means "pointer to int". In other words, we are saying that `p` is a pointer to int, which means that `p` will hold the memory address of an integer. The zero value of pointers is `nil` because a `nil` pointer holds no memory address.
+
+### Getting a pointer to a variable
+
+To find the memory address of the value of a variable, we can use the `&` operator. 
+For example, if we want to find and store the member address of variable `a` in the pointer `p`, we can do the following:
+
+```go
+var a int
+a = 2
+
+var p *int
+p = &a // the variable 'p' will hold the memory address of 'a'
+```
+
+### Accessing the value via a pointer (dereferencing)
+
+When we have a pointer, we might want to know the value stored in the memory address to which it points. We can do this using the `*` operator:
+
+```go
+var a int 
+a = 2
+
+var p *int 
+p = &a // the variable 'p' will hold the memory address of 'a'
+
+var b int
+b = *p // b == 2
+```
+
+The operation `*p` fetches the value stored at the memory address stored in `p`. This operation is often called "dereferencing".
+
+We can also use the derefering operator to assign a new value to the memory address referenced by the pointer:
+
+```go
+var a int
+a = 2   // declare int variable 'a' and assign it the value of 2
+
+var pa *int
+pa = &a          // 'pa' now refers to the memory address of 'a'  
+*pa = *pa + 2    // increment by 2 the value at memory address 'pa'
+
+fmt.Println(a)   // Output: 4
+                 // 'a' will have the new value that was changed via the pointer!
+```
+
+Assigning to `*pa` will change the value stored at the memory address `pa` holds. Since `pa` holds the memory address of `a`, by assigning to `*pa` we are effectively changing the value of `a`! 
+
+A note of caution however: always check if a pointer is not `nil` before dereferencing. Dereferencing a `nil` pointer will make the program crash at runtime!
```

Comment:
I think it is very important to show the actual error message here so people can recognize it later when it happens to them.
```suggestion
A note of caution however: always check if a pointer is not `nil` before dereferencing. Dereferencing a `nil` pointer will make the program crash at runtime!

```go
var p *int // p is nil initially
fmt.Println(*p)
// panic: runtime error: invalid memory address or nil pointer dereference
```